decentraland/unity-explorer
# fix: sdk avatar shape

### WHY

Currently the `AvatarShape` SDK component (non-player avatars) is not working at all.

Issue: https://github.com/decentraland/unity-explorer/issues/2645

### WHAT

- For the global entity representing the SDK avatar (non-player avatar), replaced the `TransformComponent` by the now needed `CharacterTransform`
- Fixed SDK avatars cleanup
- Updated test to cover the new need
- Minor code cleanup
- Edited nametag systems to support non player avatars

### QA Steps

1. Download the build from this PR
2. Open it and go to `-129,-77`
3. Click on both cubes and confirm the avatar shapes appear (it may take a bit of time until they appear).

https://github.com/user-attachments/assets/27cb058f-08f8-4e1a-83aa-b84fe887b0e3

4. Confirm that one avatar has a name tag "Dagon", and the other doesn't have a name tag. That's configured from the scene code.
5. Check that the missing avatars mentioned in https://github.com/decentraland/unity-explorer/issues/2645 are now visible.
6. Do a smoke test regarding name tags: make sure player avatar name tags keep working as expected (claimed names and unclaimed ones)
